server/src/main/java/org/jetbrains/bsp/bazel/server/sync/BazelProjectMapper.java

```diff
@@ -145,4 +148,15 @@ private Map<URI, Label> buildReverseSourceMapping(Seq<Module> modules) {
         });
     return HashMap.ofAll(output);
   }
+
+  private Map<String, String> environmentItem(TargetInfo target) {
+    var output = targetInfoToMapOfStrings(target);
+    var targetEnv = target.getEnvMap();
+    return output.merge((Map<? extends String, ? extends String>) targetEnv);
```

Comment:
it's a wrong casting - in general u want to avoid `(type)` casting, U can just wrap it with `HashMap.ofAll(....)` - it's one of the reasons why tests are failing